File: BOJ/Kotlin/그리디 알고리즘/13305.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val N = readLine().toInt()
    val railArr = readLine().split(' ').map { it.toLong() }
    val priceArr = readLine().split(' ').map { it.toLong() }
    var res = 0L
    var now = 1000000001L

    for (i in 0 until N-1){
        if (priceArr[i] < now) now = priceArr[i]
        res += now * railArr[i]
    }

    println(res)
}

File: BOJ/Python/그리디 알고리즘/13305.py
N = int(input())
railArr = list(map(int, input().split()))
priceArr = list(map(int, input().split()))
res = 0

now = float('inf')

for i in range(N-1):
    if priceArr[i] < now:
        now = priceArr[i]

    res += now * railArr[i]

print(res)
